core/src/main/scala/cell/HandlerPool.scala

```diff
@@ -277,10 +298,10 @@ class HandlerPool(parallelism: Int = 8, unhandledExceptionHandler: Throwable =>
       val state = poolState.get()
       if (state.submittedTasks > i) {
         handlersToRun = None
-        val newState = new PoolState(state.handlers, state.submittedTasks - i)
+        val newState = new PoolState(state.quiescenceHandlers, state.quiescenceCellHandlers, state.submittedTasks - i)
         success = poolState.compareAndSet(state, newState)
       } else if (state.submittedTasks == 1) {
-        handlersToRun = Some(state.handlers)
+        handlersToRun = Some(state.quiescenceHandlers)
```

Comment:
This appears to be missing `state.quiescenceCellHandlers`. Otherwise, quiescence cell handlers are only run upon registration if the pool is already quiescent at that point.

Also, it would be good to add a test which would catch this issue (i.e., register quiescence cell handler when the pool is definitely not quiescent, and verify the handler runs eventually).